src/discovery/Akka.Discovery.AwsApi/Ec2/Ec2CredentialProvider.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System.Linq;
+using Akka.Actor;
+using Akka.Event;
+using Amazon.Runtime;
+using Amazon.Util;
+
+namespace Akka.Discovery.AwsApi.Ec2
+{
+    public abstract class Ec2CredentialProvider
+    {
+        public abstract AWSCredentials ClientCredentials { get; }
+    }
+
+    public sealed class DefaultEc2CredentialProvider : Ec2CredentialProvider
+    {
+        public override AWSCredentials ClientCredentials { get; } = new AnonymousAWSCredentials();
+    }
+
+    public sealed class Ec2InstanceMetadataCredentialProvider : Ec2CredentialProvider
```

Comment:
Any reason this shouldn't be the default? Anonymous will almost always fail for these calls.

Reply:
good point, i'll change that

Reply:
Done

Reply:
Might want to rename the class too (i.e. change it to `anonymous` something something), just to clear it up, and add an XML-DOC comment indicating that the instance metadata is the default. Then we should be good to go.

Reply:
Changed the name of the class from "Default" to "Anonymous" to avoid confusion